IndicesRequestCacheIT#testQueryRewrite fails

IndicesRequestCacheIT#testQueryRewrite fails with an unexpected cache count.

```
FAILURE 7.39s J1 | IndicesRequestCacheIT.testQueryRewrite <<< FAILURES!
   > Throwable #1: java.lang.AssertionError: 
   > Expected: <3L>
   >      but: was <2L>
   > 	at __randomizedtesting.SeedInfo.seed([7B6066660D23E3B1:8A7EC1C3E5F6C580]:0)
   > 	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
   > 	at org.elasticsearch.indices.IndicesRequestCacheIT.testQueryRewrite(IndicesRequestCacheIT.java:131)
   > 	at java.lang.Thread.run(Thread.java:748)
```

This does not reproduce locally for me.

```
./gradlew :server:integTest \
  -Dtests.seed=7B6066660D23E3B1 \
  -Dtests.class=org.elasticsearch.indices.IndicesRequestCacheIT \
  -Dtests.method="testQueryRewrite" \
  -Dtests.security.manager=true \
  -Dtests.locale=en \
  -Dtests.timezone=Europe/Gibraltar \
  -Dcompiler.java=10 \
  -Druntime.java=8
```

CI: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+periodic/6845/console
Log: [testQueryRewrite.txt](https://github.com/elastic/elasticsearch/files/2284988/testQueryRewrite.txt)

Although on a different test method in the same test class, I think this is the same failure: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+matrix-java-periodic/ES_BUILD_JAVA=java11,ES_RUNTIME_JAVA=java11,nodes=virtual&&linux/265/console . I can't repro either.

```
07:10:35   1> [2018-08-28T08:10:30,847][INFO ][o.e.i.IndicesRequestCacheIT] [testQueryRewriteDatesWithNow] after test
07:10:35 FAILURE 2.58s J1 | IndicesRequestCacheIT.testQueryRewriteDatesWithNow <<< FAILURES!
07:10:35    > Throwable #1: java.lang.AssertionError: 
07:10:35    > Expected: <2L>
07:10:35    >      but: was <1L>
07:10:35    > 	at __randomizedtesting.SeedInfo.seed([FACB28E405660D99:4EC92144B233E176]:0)
07:10:35    > 	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
07:10:35    > 	at org.elasticsearch.indices.IndicesRequestCacheIT.testQueryRewriteDatesWithNow(IndicesRequestCacheIT.java:349)
07:10:35    > 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
07:10:35    > 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
07:10:35    > 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
07:10:35    > 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
07:10:35    > 	at java.base/java.lang.Thread.run(Thread.java:834)
```

Have a new instance of this: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+intake/255/console

```
06:57:43   1> [2018-11-20T14:57:38,969][INFO ][o.e.i.IndicesRequestCacheIT] [testQueryRewrite] [IndicesRequestCacheIT#testQueryRewrite]: cleaning up after test
06:57:43   1> [2018-11-20T14:57:39,018][INFO ][o.e.c.m.MetaDataDeleteIndexService] [node_sm2] [index/j7ReR8i1R1OIrDAmbZH9RA] deleting index
06:57:43   1> [2018-11-20T14:57:39,113][INFO ][o.e.c.m.MetaDataIndexTemplateService] [node_sm2] removing template [random_index_template]
06:57:43   1> [2018-11-20T14:57:39,131][INFO ][o.e.i.IndicesRequestCacheIT] [testQueryRewrite] [IndicesRequestCacheIT#testQueryRewrite]: cleaned up after test
06:57:43   1> [2018-11-20T14:57:39,131][INFO ][o.e.i.IndicesRequestCacheIT] [testQueryRewrite] after test
06:57:43 FAILURE 13.2s J6 | IndicesRequestCacheIT.testQueryRewrite <<< FAILURES!
06:57:43    > Throwable #1: java.lang.AssertionError: expected:<[6, 9, 0]> but was:<[5, 10, 0]>
06:57:43    > 	at __randomizedtesting.SeedInfo.seed([48E304082D784C6:F59097E56A02A2F7]:0)
06:57:43    > 	at org.elasticsearch.indices.IndicesRequestCacheIT.assertCacheState(IndicesRequestCacheIT.java:422)
06:57:43    > 	at org.elasticsearch.indices.IndicesRequestCacheIT.testQueryRewrite(IndicesRequestCacheIT.java:144)
06:57:43    > 	at java.lang.Thread.run(Thread.java:748)
```

We have a couple more errors this weekend:

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.x+multijob-unix-compatibility/os=centos/80/console

```
java.lang.AssertionError: 
Expected: <1L>
     but: was <0L>
	at __randomizedtesting.SeedInfo.seed([43BC570BE27C13A6:F7BE5EAB5529FF49]:0)
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.junit.Assert.assertThat(Assert.java:956)
	at org.junit.Assert.assertThat(Assert.java:923)
	at org.elasticsearch.indices.IndicesRequestCacheIT.testQueryRewriteDatesWithNow(IndicesRequestCacheIT.java:319)
```

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.x+multijob-unix-compatibility/os=fedora/79/console

```
ava.lang.AssertionError: 
Expected: <3L>
     but: was <2L>
	at __randomizedtesting.SeedInfo.seed([9A2FA9AA40878B16:6B310E0FA852AD27]:0)
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.junit.Assert.assertThat(Assert.java:956)
	at org.junit.Assert.assertThat(Assert.java:923)
	at org.elasticsearch.indices.IndicesRequestCacheIT.testQueryRewrite(IndicesRequestCacheIT.java:130)
```

We had another failure last night, this time in a different test method on the same class:
```
java.lang.AssertionError: expected:<[1, 1, 0]> but was:<[0, 2, 0]>
	at __randomizedtesting.SeedInfo.seed([837DAA566443EF4B:C4AE44D1149A5879]:0)
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:118)
	at org.junit.Assert.assertEquals(Assert.java:144)
	at org.elasticsearch.indices.IndicesRequestCacheIT.assertCacheState(IndicesRequestCacheIT.java:424)
	at org.elasticsearch.indices.IndicesRequestCacheIT.testCacheWithFilteredAlias(IndicesRequestCacheIT.java:406)
```

Link to build: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+internalClusterTest/1678/console

Reproduce line (doesn't reproduce, of course):
```
REPRODUCE WITH: ./gradlew :server:integTest \
  -Dtests.seed=837DAA566443EF4B \
  -Dtests.class=org.elasticsearch.indices.IndicesRequestCacheIT \
  -Dtests.method="testCacheWithFilteredAlias" \
  -Dtests.security.manager=true \
  -Dtests.locale=sv-SE \
  -Dtests.timezone=Europe/Amsterdam \
  -Dcompiler.java=11 \
  -Druntime.java=8
```

Relevant part of the logs for reader versions:
```
01:57:13   1> [2019-02-22T02:57:07,105][INFO ][o.e.p.PluginsService     ] [testCacheWithFilteredAlias] loaded plugin [org.elasticsearch.transport.nio.MockNioTransportPlugin]
01:57:13   1> [2019-02-22T02:57:07,745][TRACE][o.e.i.IndicesRequestCache] [node_s2] Cache miss for reader version [7] and request:
01:57:13   1>  Shard: [index][0]
01:57:13   1> Source:
01:57:13   1> {"size":0,"query":{"range":{"created_at":{"from":null,"to":null,"include_lower":true,"include_upper":true,"boost":1.0}}}}
01:57:13   1> [2019-02-22T02:57:07,808][TRACE][o.e.i.IndicesRequestCache] [node_s2] Cache miss for reader version [9] and request:
01:57:13   1>  Shard: [index][0]
01:57:13   1> Source:
01:57:13   1> {"size":0,"query":{"range":{"created_at":{"from":null,"to":null,"include_lower":true,"include_upper":true,"boost":1.0}}}}
01:57:13   1> [2019-02-22T02:57:07,831][INFO ][o.e.i.IndicesRequestCacheIT] [testCacheWithFilteredAlias] [IndicesRequestCacheIT#testCacheWithFilteredAlias]: cleaning up after test
01:57:13   1> [2019-02-22T02:57:07,863][INFO ][o.e.c.m.MetaDataDeleteIndexService] [node_s1] [index/tUjeiKVrRU2oaImpCPeTvQ] deleting index
01:57:13   1> [2019-02-22T02:57:07,932][INFO ][o.e.c.m.MetaDataIndexTemplateService] [node_s1] removing template [random_index_template]
01:57:13   1> [2019-02-22T02:57:07,954][INFO ][o.e.i.IndicesRequestCacheIT] [testCacheWithFilteredAlias] [IndicesRequestCacheIT#testCacheWithFilteredAlias]: cleaned up after test
01:57:13   1> [2019-02-22T02:57:07,954][INFO ][o.e.i.IndicesRequestCacheIT] [testCacheWithFilteredAlias] after test
01:57:13 FAILURE 2.17s J6 | IndicesRequestCacheIT.testCacheWithFilteredAlias <<< FAILURES!
01:57:13    > Throwable #1: java.lang.AssertionError: expected:<[1, 1, 0]> but was:<[0, 2, 0]>
01:57:13    > 	at __randomizedtesting.SeedInfo.seed([837DAA566443EF4B:C4AE44D1149A5879]:0)
01:57:13    > 	at org.elasticsearch.indices.IndicesRequestCacheIT.assertCacheState(IndicesRequestCacheIT.java:424)
01:57:13    > 	at org.elasticsearch.indices.IndicesRequestCacheIT.testCacheWithFilteredAlias(IndicesRequestCacheIT.java:406)
01:57:13    > 	at java.lang.Thread.run(Thread.java:748)
```

Another one in https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.8+periodic/390/console

Failure:

```
FAILURE 1.41s J7 | IndicesRequestCacheIT.testQueryRewriteDatesWithNow <<< FAILURES!
17:44:20    > Throwable #1: java.lang.AssertionError: expected:<[1, 1]> but was:<[0, 2]>
17:44:20    > 	at __randomizedtesting.SeedInfo.seed([C2622D4632A7E5E8:766024E685F20907]:0)
17:44:20    > 	at org.elasticsearch.indices.IndicesRequestCacheIT.assertCacheState(IndicesRequestCacheIT.java:418)
17:44:20    > 	at org.elasticsearch.indices.IndicesRequestCacheIT.testQueryRewriteDatesWithNow(IndicesRequestCacheIT.java:286)
17:44:20    > 	at java.lang.Thread.run(Thread.java:748)
```

It also fails on 7.x https://gradle-enterprise.elastic.co/s/tkfitlc3lhwmw

```
REPRODUCE WITH: ./gradlew ':server:integTest' --tests "org.elasticsearch.indices.IndicesRequestCacheIT.testQueryRewrite" -Dtests.seed=7353DB5A0E87365D -Dtests.security.manager=true -Dtests.locale=sl-SI -Dtests.timezone=America/Hermosillo -Dcompiler.java=13	
  2> java.lang.AssertionError: expected:<[6, 9, 0]> but was:<[5, 10, 0]>	
        at __randomizedtesting.SeedInfo.seed([7353DB5A0E87365D:824D7CFFE652106C]:0)	
        at org.junit.Assert.fail(Assert.java:88)	
        at org.junit.Assert.failNotEquals(Assert.java:834)	
        at org.junit.Assert.assertEquals(Assert.java:118)	
        at org.junit.Assert.assertEquals(Assert.java:144)	
        at org.elasticsearch.indices.IndicesRequestCacheIT.assertCacheState(IndicesRequestCacheIT.java:469)	
        at org.elasticsearch.indices.IndicesRequestCacheIT.testQueryRewrite(IndicesRequestCacheIT.java:150)
```
I'll wait for one more failure before muting on 7.x as well.

Another time, and also `testQueryRewriteDatesWithNow` fails:
```
REPRODUCE WITH: ./gradlew ':server:integTest' --tests "org.elasticsearch.indices.IndicesRequestCacheIT.testQueryRewrite" -Dtests.seed=F4D7D8FF9792A390 -Dtests.security.manager=true -Dtests.locale=ms-MY -Dtests.timezone=America/Ensenada -Dcompiler.java=13	
	
org.elasticsearch.indices.IndicesRequestCacheIT > testQueryRewrite FAILED	
    java.lang.AssertionError: expected:<[3, 7, 0]> but was:<[2, 8, 0]>	
        at __randomizedtesting.SeedInfo.seed([F4D7D8FF9792A390:5C97F5A7F4785A1]:0)	
        at org.junit.Assert.fail(Assert.java:88)	
        at org.junit.Assert.failNotEquals(Assert.java:834)	
        at org.junit.Assert.assertEquals(Assert.java:118)	
        at org.junit.Assert.assertEquals(Assert.java:144)	
        at org.elasticsearch.indices.IndicesRequestCacheIT.assertCacheState(IndicesRequestCacheIT.java:469)	
        at org.elasticsearch.indices.IndicesRequestCacheIT.testQueryRewrite(IndicesRequestCacheIT.java:141)	
	
REPRODUCE WITH: ./gradlew ':server:integTest' --tests "org.elasticsearch.indices.IndicesRequestCacheIT.testQueryRewriteDatesWithNow" -Dtests.seed=F4D7D8FF9792A390 -Dtests.security.manager=true -Dtests.locale=ms-MY -Dtests.timezone=America/Ensenada -Dcompiler.java=13	
	
org.elasticsearch.indices.IndicesRequestCacheIT > testQueryRewriteDatesWithNow FAILED	
    java.lang.AssertionError: expected:<[2, 1, 0]> but was:<[1, 2, 0]>	
        at __randomizedtesting.SeedInfo.seed([F4D7D8FF9792A390:40D5D15F20C74F7F]:0)	
        at org.junit.Assert.fail(Assert.java:88)	
        at org.junit.Assert.failNotEquals(Assert.java:834)	
        at org.junit.Assert.assertEquals(Assert.java:118)	
        at org.junit.Assert.assertEquals(Assert.java:144)	
        at org.elasticsearch.indices.IndicesRequestCacheIT.assertCacheState(IndicesRequestCacheIT.java:469)	
        at org.elasticsearch.indices.IndicesRequestCacheIT.testQueryRewriteDatesWithNow(IndicesRequestCacheIT.java:305)
```

I'm going to mute this in both master and 7.x, please feel free to revert if I'm being too aggressive.

--- a/server/src/main/java/org/elasticsearch/search/internal/ShardSearchRequest.java
+++ b/server/src/main/java/org/elasticsearch/search/internal/ShardSearchRequest.java
@@ -219,7 +219,9 @@ public class ShardSearchRequest extends TransportRequest implements IndicesReque
             out.writeOptionalString(preference);
         }
         if (out.getVersion().onOrAfter(Version.V_7_7_0)) {
-            out.writeBoolean(canReturnNullResponseIfMatchNoDocs);
+            if (asKey == false) {
+                out.writeBoolean(canReturnNullResponseIfMatchNoDocs);
+            }
             out.writeOptionalWriteable(bottomSortValues);
         }
     }

--- a/server/src/main/java/org/elasticsearch/search/internal/ShardSearchRequest.java
+++ b/server/src/main/java/org/elasticsearch/search/internal/ShardSearchRequest.java
@@ -218,10 +218,8 @@ public class ShardSearchRequest extends TransportRequest implements IndicesReque
             out.writeStringArray(indexRoutings);
             out.writeOptionalString(preference);
         }
-        if (out.getVersion().onOrAfter(Version.V_7_7_0)) {
-            if (asKey == false) {
-                out.writeBoolean(canReturnNullResponseIfMatchNoDocs);
-            }
+        if (out.getVersion().onOrAfter(Version.V_7_7_0) && asKey == false) {
+            out.writeBoolean(canReturnNullResponseIfMatchNoDocs);
             out.writeOptionalWriteable(bottomSortValues);
         }
     }

--- a/server/src/test/java/org/elasticsearch/indices/IndicesRequestCacheIT.java
+++ b/server/src/test/java/org/elasticsearch/indices/IndicesRequestCacheIT.java
@@ -101,7 +101,6 @@ public class IndicesRequestCacheIT extends ESIntegTestCase {
     @TestIssueLogging(
         value = "org.elasticsearch.indices.IndicesRequestCache:TRACE",
         issueUrl = "https://github.com/elastic/elasticsearch/issues/32827")
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/32827")
     public void testQueryRewrite() throws Exception {
         Client client = client();
         assertAcked(client.admin().indices().prepareCreate("index").setMapping("s", "type=date")
@@ -132,7 +131,7 @@ public class IndicesRequestCacheIT extends ESIntegTestCase {
             .setQuery(QueryBuilders.rangeQuery("s").gte("2016-03-19").lte("2016-03-25"))
             // to ensure that query is executed even if it rewrites to match_no_docs
             .addAggregation(new GlobalAggregationBuilder("global"))
-            .setPreFilterShardSize(Integer.MAX_VALUE).get();
+            .get();
         ElasticsearchAssertions.assertAllSuccessful(r1);
         assertThat(r1.getHits().getTotalHits().value, equalTo(7L));
         assertCacheState(client, "index", 0, 5);
@@ -140,7 +139,7 @@ public class IndicesRequestCacheIT extends ESIntegTestCase {
         final SearchResponse r2 = client.prepareSearch("index").setSearchType(SearchType.QUERY_THEN_FETCH).setSize(0)
             .setQuery(QueryBuilders.rangeQuery("s").gte("2016-03-20").lte("2016-03-26"))
             .addAggregation(new GlobalAggregationBuilder("global"))
-            .setPreFilterShardSize(Integer.MAX_VALUE).get();
+            .get();
         ElasticsearchAssertions.assertAllSuccessful(r2);
         assertThat(r2.getHits().getTotalHits().value, equalTo(7L));
         assertCacheState(client, "index", 3, 7);
@@ -148,7 +147,6 @@ public class IndicesRequestCacheIT extends ESIntegTestCase {
         final SearchResponse r3 = client.prepareSearch("index").setSearchType(SearchType.QUERY_THEN_FETCH).setSize(0)
             .setQuery(QueryBuilders.rangeQuery("s").gte("2016-03-21").lte("2016-03-27"))
             .addAggregation(new GlobalAggregationBuilder("global"))
-            .setPreFilterShardSize(Integer.MAX_VALUE)
             .get();
         ElasticsearchAssertions.assertAllSuccessful(r3);
         assertThat(r3.getHits().getTotalHits().value, equalTo(7L));
@@ -224,29 +222,35 @@ public class IndicesRequestCacheIT extends ESIntegTestCase {
 
         assertCacheState(client, "index", 0, 0);
 
-        final SearchResponse r1 = client.prepareSearch("index").setSearchType(SearchType.QUERY_THEN_FETCH).setSize(0)
-                .setQuery(QueryBuilders.rangeQuery("d").gte("2013-01-01T00:00:00").lte("now"))
-                .get();
+        final SearchResponse r1 = client.prepareSearch("index").setSearchType(SearchType.QUERY_THEN_FETCH)
+            .setSize(0)
+            .setQuery(QueryBuilders.rangeQuery("d").gte("2013-01-01T00:00:00").lte("now"))
+            // to ensure that query is executed even if it rewrites to match_no_docs
+            .addAggregation(new GlobalAggregationBuilder("global"))
+            .get();
         ElasticsearchAssertions.assertAllSuccessful(r1);
         assertThat(r1.getHits().getTotalHits().value, equalTo(9L));
         assertCacheState(client, "index", 0, 1);
 
-        final SearchResponse r2 = client.prepareSearch("index").setSearchType(SearchType.QUERY_THEN_FETCH).setSize(0)
-                .setQuery(QueryBuilders.rangeQuery("d").gte("2013-01-01T00:00:00").lte("now"))
-                .get();
+        final SearchResponse r2 = client.prepareSearch("index").setSearchType(SearchType.QUERY_THEN_FETCH)
+            .setSize(0)
+            .setQuery(QueryBuilders.rangeQuery("d").gte("2013-01-01T00:00:00").lte("now"))
+            .addAggregation(new GlobalAggregationBuilder("global"))
+            .get();
         ElasticsearchAssertions.assertAllSuccessful(r2);
         assertThat(r2.getHits().getTotalHits().value, equalTo(9L));
         assertCacheState(client, "index", 1, 1);
 
-        final SearchResponse r3 = client.prepareSearch("index").setSearchType(SearchType.QUERY_THEN_FETCH).setSize(0)
-                .setQuery(QueryBuilders.rangeQuery("d").gte("2013-01-01T00:00:00").lte("now"))
-                .get();
+        final SearchResponse r3 = client.prepareSearch("index").setSearchType(SearchType.QUERY_THEN_FETCH)
+            .setSize(0)
+            .setQuery(QueryBuilders.rangeQuery("d").gte("2013-01-01T00:00:00").lte("now"))
+            .addAggregation(new GlobalAggregationBuilder("global"))
+            .get();
         ElasticsearchAssertions.assertAllSuccessful(r3);
         assertThat(r3.getHits().getTotalHits().value, equalTo(9L));
         assertCacheState(client, "index", 2, 1);
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/32827")
     public void testQueryRewriteDatesWithNow() throws Exception {
         Client client = client();
         Settings settings = Settings.builder().put(IndicesRequestCache.INDEX_CACHE_REQUEST_ENABLED_SETTING.getKey(), true)
@@ -284,8 +288,10 @@ public class IndicesRequestCacheIT extends ESIntegTestCase {
         assertCacheState(client, "index-2", 0, 0);
         assertCacheState(client, "index-3", 0, 0);
 
-        final SearchResponse r1 = client.prepareSearch("index-*").setSearchType(SearchType.QUERY_THEN_FETCH).setSize(0)
-                .setQuery(QueryBuilders.rangeQuery("d").gte("now-7d/d").lte("now")).get();
+        final SearchResponse r1 = client.prepareSearch("index-*")
+            .setSearchType(SearchType.QUERY_THEN_FETCH)
+            .setSize(0)
+            .setQuery(QueryBuilders.rangeQuery("d").gte("now-7d/d").lte("now")).get();
         ElasticsearchAssertions.assertAllSuccessful(r1);
         assertThat(r1.getHits().getTotalHits().value, equalTo(8L));
         assertCacheState(client, "index-1", 0, 1);
@@ -295,16 +301,22 @@ public class IndicesRequestCacheIT extends ESIntegTestCase {
         // cache miss or cache hit since queries containing now can't be cached
         assertCacheState(client, "index-3", 0, 0);
 
-        final SearchResponse r2 = client.prepareSearch("index-*").setSearchType(SearchType.QUERY_THEN_FETCH).setSize(0)
-                .setQuery(QueryBuilders.rangeQuery("d").gte("now-7d/d").lte("now")).get();
+        final SearchResponse r2 = client.prepareSearch("index-*")
+            .setSearchType(SearchType.QUERY_THEN_FETCH)
+            .setSize(0)
+            .setQuery(QueryBuilders.rangeQuery("d").gte("now-7d/d").lte("now"))
+            .get();
         ElasticsearchAssertions.assertAllSuccessful(r2);
         assertThat(r2.getHits().getTotalHits().value, equalTo(8L));
         assertCacheState(client, "index-1", 1, 1);
         assertCacheState(client, "index-2", 1, 1);
         assertCacheState(client, "index-3", 0, 0);
 
-        final SearchResponse r3 = client.prepareSearch("index-*").setSearchType(SearchType.QUERY_THEN_FETCH).setSize(0)
-                .setQuery(QueryBuilders.rangeQuery("d").gte("now-7d/d").lte("now")).get();
+        final SearchResponse r3 = client.prepareSearch("index-*")
+            .setSearchType(SearchType.QUERY_THEN_FETCH)
+            .setSize(0)
+            .setQuery(QueryBuilders.rangeQuery("d").gte("now-7d/d").lte("now"))
+            .get();
         ElasticsearchAssertions.assertAllSuccessful(r3);
         assertThat(r3.getHits().getTotalHits().value, equalTo(8L));
         assertCacheState(client, "index-1", 2, 1);

